File: jlpt/native_ios/Sources/Presentation/Study/Components/StudyCardContent.swift
import SwiftUI

struct VocabularyFront: View {
    let item: Vocabulary
    var body: some View {
        VStack(spacing: 8) {
            Text(item.word).font(.system(size: 40))
            Text(item.reading).font(.title3)
        }
        .multilineTextAlignment(.center)
    }
}

struct VocabularyBack: View {
    let item: Vocabulary
    var body: some View {
        VStack(spacing: 8) {
            Text(item.meaningVi).font(.title3.weight(.semibold))
            if let partOfSpeech = item.partOfSpeech {
                Text(partOfSpeech).font(.subheadline).foregroundStyle(.secondary)
            }
            if let example = item.exampleSentence {
                Text(example).font(.body)
            }
            if let meaning = item.exampleMeaning {
                Text(meaning).font(.body)
            }
        }
        .multilineTextAlignment(.center)
    }
}

struct KanjiFront: View {
    let item: Kanji
    var body: some View {
        Text(item.character).font(.system(size: 72))
    }
}

struct KanjiBack: View {
    let item: Kanji
    var body: some View {
        VStack(spacing: 8) {
            Text(item.meaningVi).font(.title3.weight(.semibold))
            if let onyomi = item.onyomi { Text("Onyomi: \(onyomi)").font(.body) }
            if let kunyomi = item.kunyomi { Text("Kunyomi: \(kunyomi)").font(.body) }
            if let words = item.exampleWords { Text(words).font(.body) }
        }
        .multilineTextAlignment(.center)
    }
}

struct GrammarFront: View {
    let item: Grammar
    var body: some View {
        Text(item.pattern).font(.title.weight(.bold)).multilineTextAlignment(.center)
    }
}

struct GrammarBack: View {
    let item: Grammar
    var body: some View {
        VStack(spacing: 8) {
            Text(item.meaningVi).font(.title3.weight(.semibold))
            if let note = item.usageNote { Text(note).font(.body) }
            if let example = item.exampleSentence { Text(example).font(.body) }
            if let meaning = item.exampleMeaning { Text(meaning).font(.body) }
        }
        .multilineTextAlignment(.center)
    }
}

File: jlpt/native_android/app/src/main/java/com/jlpt/android/presentation/study/components/StudyCardContent.kt
package com.jlpt.android.presentation.study.components

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.jlpt.android.domain.model.Grammar
import com.jlpt.android.domain.model.Kanji
import com.jlpt.android.domain.model.Vocabulary

@Composable
fun VocabularyFront(item: Vocabulary) {
    Column(horizontalAlignment = Alignment.CenterHorizontally, verticalArrangement = Arrangement.spacedBy(8.dp)) {
        Text(item.word, fontSize = 40.sp, textAlign = TextAlign.Center)
        Text(item.reading, style = MaterialTheme.typography.titleLarge, textAlign = TextAlign.Center)
    }
}

@Composable
fun VocabularyBack(item: Vocabulary) {
    Column(horizontalAlignment = Alignment.CenterHorizontally, verticalArrangement = Arrangement.spacedBy(8.dp)) {
        Text(item.meaningVi, style = MaterialTheme.typography.titleLarge, textAlign = TextAlign.Center)
        item.partOfSpeech?.let { Text(it, style = MaterialTheme.typography.labelLarge) }
        item.exampleSentence?.let {
            Text(it, textAlign = TextAlign.Center, style = MaterialTheme.typography.bodyLarge)
        }
        item.exampleMeaning?.let {
            Text(it, textAlign = TextAlign.Center, style = MaterialTheme.typography.bodyLarge)
        }
    }
}

@Composable
fun KanjiFront(item: Kanji) {
    Text(item.character, fontSize = 72.sp, textAlign = TextAlign.Center)
}

@Composable
fun KanjiBack(item: Kanji) {
    Column(horizontalAlignment = Alignment.CenterHorizontally, verticalArrangement = Arrangement.spacedBy(8.dp)) {
        Text(item.meaningVi, style = MaterialTheme.typography.titleLarge, textAlign = TextAlign.Center)
        item.onyomi?.let { Text("Onyomi: $it", style = MaterialTheme.typography.bodyLarge) }
        item.kunyomi?.let { Text("Kunyomi: $it", style = MaterialTheme.typography.bodyLarge) }
        item.exampleWords?.let { Text(it, textAlign = TextAlign.Center, style = MaterialTheme.typography.bodyLarge) }
    }
}

@Composable
fun GrammarFront(item: Grammar) {
    Text(item.pattern, style = MaterialTheme.typography.headlineMedium, textAlign = TextAlign.Center)
}

@Composable
fun GrammarBack(item: Grammar) {
    Column(horizontalAlignment = Alignment.CenterHorizontally, verticalArrangement = Arrangement.spacedBy(8.dp)) {
        Text(item.meaningVi, style = MaterialTheme.typography.titleLarge, textAlign = TextAlign.Center)
        item.usageNote?.let { Text(it, textAlign = TextAlign.Center, style = MaterialTheme.typography.bodyLarge) }
        item.exampleSentence?.let { Text(it, textAlign = TextAlign.Center, style = MaterialTheme.typography.bodyLarge) }
        item.exampleMeaning?.let { Text(it, textAlign = TextAlign.Center, style = MaterialTheme.typography.bodyLarge) }
    }
}
